airbyte-cdk/python/airbyte_cdk/sources/declarative/async_job/job_orchestrator.py

```diff
@@ -226,7 +224,12 @@ def create_and_get_completed_partitions(self) -> Iterable[AsyncPartition]:
             Each partition is wrapped in an Optional, allowing for None values.
         """
         while True:
-            self._start_jobs()
+            try:
+                self._start_jobs()
+            except self._exceptions_to_break_on as e:
+                raise e
+            except AirbyteTracedException as e:
+                LOGGER.error(f"Failed to start the Job: {e.message} , {e.internal_message}")
```

Comment:
Can we add tests to validate this behavior? What I'm afraid might happen here is that the partitions will still be in the running partitions and therefore we will try to start them again next iteration potentially leading to an infinite loop. The tests I would expect would be something like this

### Given exception to break on then raise exception
* have self._job_repository.start raise an exception that we should break on
* call `create_and_get_completed_partitions`
* with pytest.raises: <...>

### Given any other exception then skip this slice and continue looping on the other>
* have multiple slices (let's say 2)
* have self._job_repository.start raise an exception that we should *not* break on on one of the slices
* call `create_and_get_completed_partitions`
* have a number of partitions returned equals to the number of slices minus one

If you see any other meaningful tests, feel free to add them

Reply:
added